org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeNewFormatterSetup.xtend

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.ide.tests.testlanguage.ide
+
+import com.google.inject.Binder
+import com.google.inject.Guice
+import com.google.inject.Injector
+import com.google.inject.name.Names
+import org.eclipse.emf.ecore.EPackage
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.Constants
+import org.eclipse.xtext.formatting2.IFormatter2
+import org.eclipse.xtext.ide.tests.testlanguage.TestLanguageRuntimeModule
+import org.eclipse.xtext.ide.tests.testlanguage.TestLanguageStandaloneSetup
+import org.eclipse.xtext.ide.tests.testlanguage.formatting2.TestLanguageFormatter
+import org.eclipse.xtext.ide.tests.testlanguage.testLanguage.TestLanguagePackage
+import org.eclipse.xtext.resource.IResourceFactory
+import org.eclipse.xtext.resource.IResourceServiceProvider
+
+/**
+ * Initialization support for running Xtext languages without Equinox extension registry.
+ */
+class TestLanguageIdeNewFormatterSetup extends TestLanguageStandaloneSetup {
```

Comment:
could you make the formatter impl part of the TestLanguage? Then this special setup should not be necessary anymore.